pallets/runtime/tests/src/corporate_actions_test.rs

```diff
@@ -118,31 +127,103 @@ fn transfer_caa(ticker: Ticker, from: User, to: User) -> DispatchResult {
     Identity::accept_authorization(to.signer(), auth_id)
 }
 
+type CAResult = Result<CorporateAction, DispatchError>;
+
 fn init_ca(
     owner: User,
     ticker: Ticker,
     kind: CAKind,
-    date: Option<Moment>,
+    date: Option<RecordDateSpec>,
     details: String,
     targets: Option<TargetIdentities>,
     default_wht: Option<Tax>,
     wht: Option<Vec<(IdentityId, Tax)>>,
-) -> Result<CorporateAction, DispatchError> {
+) -> CAResult {
     let id = CA::ca_id_sequence(ticker);
     let sig = owner.signer();
     let details = CADetails(details.as_bytes().to_vec());
-    let date = date.map(RecordDateSpec::Scheduled);
     CA::initiate_corporate_action(sig, ticker, kind, date, details, targets, default_wht, wht)?;
     Ok(CA::corporate_actions(ticker, id).unwrap())
 }
 
+fn basic_ca(
+    owner: User,
+    ticker: Ticker,
+    targets: Option<TargetIdentities>,
+    default_wht: Option<Tax>,
+    wht: Option<Vec<(IdentityId, Tax)>>,
+) -> CAResult {
+    init_ca(
+        owner,
+        ticker,
+        CAKind::Other,
+        None,
+        <_>::default(),
+        targets,
+        default_wht,
+        wht,
+    )
+}
+
+fn dated_ca(owner: User, ticker: Ticker, kind: CAKind, rd: Option<RecordDateSpec>) -> CAResult {
+    init_ca(owner, ticker, kind, rd, <_>::default(), None, None, None)
+}
+
+fn moment_ca(owner: User, ticker: Ticker, kind: CAKind, rd: Option<Moment>) -> CAResult {
+    dated_ca(owner, ticker, kind, rd.map(RecordDateSpec::Scheduled))
+}
+
+fn set_schedule_complexity() {
+    Timestamp::set_timestamp(1);
+    assert_ok!(Checkpoint::set_schedules_max_complexity(root(), 1000));
+}
+
+fn next_ca_id(ticker: Ticker) -> CAId {
+    let local_id = CA::ca_id_sequence(ticker);
+    CAId { ticker, local_id }
+}
+
+const TRANGE: BallotTimeRange = BallotTimeRange {
+    start: 3000,
+    end: 4000,
+};
+
+#[derive(Clone, Eq, PartialEq, Default, Debug)]
```

Comment:
Why do we need these in tests 🤔

Reply:
`Eq, PartialEq, Debug` are used by `assert_eq`s, `Clone` is used for FRU in various places, and `Default` is used to test against "initial state". The data type exists to reduce duplication and to more rigorously test the state space instead of individual components.